### ISSUE TITLE
是否支持VLM 输出logprobs

### ISSUE THREAD

**Issue by hxyghostor** [InternLM/lmdeploy#3243]
Can VLLM(Qwen2-vl-2b-4bit) support logprobs output? I always get None

![Image](https://github.com/user-attachments/assets/cec61a06-ed86-4324-8206-64b1077c009b)

**Comment by hxyghostor** [comment]
server

lmdeploy serve api_server /question_classification/qwen2-vl-2b-4bit --server-port $PORT0 --backend turbomind --model-format awq --enable-prefix-caching --quant-policy 8

client
import requests
import base64

api_url = f"http://localhost:10516/v1/chat/completions"



image_path = ""
with open(image_path, "rb") as f:
            encoded_image = base64.b64encode(f.read())
encoded_image_text = encoded_image.decode("utf-8")
base64_qwen = f"data:image;base64,{encoded_image_text}"
data = {
            "model" : "/question_classification/qwen2-vl-2b-4bit",
            "messages":[
                {"role": "system", "content": "You are a helpful assistant."},
                {
                    "role": "user",
                    "content": [
                        {
                            "type": "image_url",
                            "image_url": {
                                "url": base64_qwen
                            },
                        },
                        {"type": "text", 
                         "text": ""},
                    ],
                },
            ],
            "logprobs": True,
            "top_logprobs": 3
        }

chat_response = requests.post(api_url, json=data).json()
print("Chat response:", chat_response)

**Comment by lvhan028** [comment]
Sorry, the pytorch engine hasn't supported output `logprob` yet.

**Comment by lvhan028** [comment]
Probably at the end of this month. I am reviewing the #3164. Once it is merged to main branch, @irexyc can submit the implementation of qwen2-vl and qwen2.5-vl

**Comment by hxyghostor** [comment]
OK，thanks for your reply.

**Comment by moyans** [comment]
> Probably at the end of this month. I am reviewing the [#3164](https://github.com/InternLM/lmdeploy/pull/3164). Once it is merged to main branch, [@irexyc](https://github.com/irexyc) can submit the implementation of qwen2-vl and qwen2.5-vl

May I ask if it is already supported?

**Comment by zhangquanchen** [comment]
The **logprobs is None** when I using Qwen2.5-vl as follow, haven't it support now? or how can I get the value?

[Response(text='否', generate_token_len=1, input_token_len=767, finish_reason='length', token_ids=[32376], logprobs=None, logits=None, last_hidden_state=None, index=0)]

**Comment by hekaijie123** [comment]
在Internvl2.5模型中，使用--backend  turbomind，输出的logprob也为None
for item in api_client.chat_completions_v1(model=model_name,
            temperature=0,
            max_tokens=length,   messages=messages,logprobs=True):
